exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/FunctionsTests.kt

```diff
@@ -59,6 +61,57 @@ class FunctionsTests : DatabaseTestsBase() {
         }
     }
 
+    @Test
+    fun testCalc04() {
+        withCitiesAndUsers { cities, users, userData ->
+            val adminFlag = DMLTestsData.Users.Flags.IS_ADMIN
+            val admin = Expression.build { AndBitOp(users.flags, intParam(adminFlag), IntegerColumnType()) eq adminFlag }
+            val r = users.slice(users.id, admin).selectAll().orderBy(users.id).toList()
+            assertEquals(5, r.size)
+            val trueType = when (currentDialectTest) {
+                is MysqlDialect -> 1L
+                is SQLiteDialect -> 1
+                else -> true
+            }
+            val falseType = when (currentDialectTest) {
+                is MysqlDialect -> 0L
+                is SQLiteDialect -> 0
+                else -> false
+            }
+            assertEquals(falseType, r[0][admin])
+            assertEquals(trueType, r[1][admin])
+            assertEquals(falseType, r[2][admin])
+            assertEquals(trueType, r[3][admin])
+            assertEquals(falseType, r[4][admin])
```

Comment:
This seemed quite weird to me when testing, although the type of the `admin` expression is `EqOp` = `Op<Boolean>`, the `get` returned the internal database representations for the dialects which a workaround has now been applied for. Might be a bug (or as misunderstanding on my side).

Reply:
It seems this still doesn't work as intended, but I might be missing something.

Reply:
You are right, the problem was in how Exposed process Op<Boolean> on read from DB.
Now it was fixed in the master, please merge and recheck.